prusti-viper/src/encoder/encoder.rs

```diff
@@ -1282,7 +1282,12 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
             let procedure = self.env.get_procedure(wrapper_def_id);
             let pure_function_encoder =
                 PureFunctionEncoder::new(self, proc_def_id, procedure.get_mir(), false);
-            let function = if self.is_trusted(proc_def_id) {
+            let function = if let Some(predicate_body) = self.get_predicate_body(proc_def_id) {
+                let predicate_body_encoded = self.encode_assertion(predicate_body, procedure.get_mir(), None, &[], None, false, None, ErrorCtxt::GenericExpression)
+                    .run_if_err(cleanup)?;
```

Comment:
*  I think that `targets_are_values` should be `true` and that you should provide `target_args` ([here](https://github.com/viperproject/prusti-dev/blob/e207038b94cb8e5499b309de7ec52765b6e2404b/prusti-viper/src/encoder/pure_function_encoder.rs#L289-L293) is an example of how to build them). See [here](https://github.com/viperproject/prusti-dev/blob/e207038b94cb8e5499b309de7ec52765b6e2404b/prusti-viper/src/encoder/spec_encoder.rs#L32-L61) for a description of the arguments.
* I would prefer to move the `get_predicate_body` and the construction of `predicate_body_encoded` into the `encode_predicate_function` function, to keep `encoder.rs` clean.
* This line is very long; consider placing one argument per line.